bundles/org.pitest.pitclipse.ui/src/org/pitest/pitclipse/ui/view/PitView.java

```diff
@@ -18,28 +18,75 @@
 import org.eclipse.swt.browser.Browser;
 import org.eclipse.swt.widgets.Composite;
+import org.eclipse.ui.IActionBars;
 import org.eclipse.ui.part.ViewPart;
 import org.pitest.pitclipse.ui.utils.PitclipseUiUtils;
 
 /**
  * A view displaying the HTML report generated by PIT.
  */
 public class PitView extends ViewPart implements SummaryView {
+    public static final String VIEW_ID = "org.pitest.pitclipse.ui.view.PitView";
+    public static final String BLANK_PAGE = "about:blank";
+    public static final String BACK_BUTTON_TEXT = "<";
+    public static final String HOME_BUTTON_TEXT = "Home";
+    public static final String FORWARD_BUTTON_TEXT = ">";
     private Browser browser = null;
+    private String homeUrlString = null;
 
     @Override
     public synchronized void createPartControl(Composite parent) {
-        PitclipseUiUtils.executeSafely
-            (() -> {
-                browser = new Browser(parent, SWT.NONE);
-                browser.addProgressListener(new PitUiUpdatePublisher(browser));
-            },
-            parent,
-            "Failed to create the Pit Summary view",
-            "Browser cannot be initialized.");
+        PitclipseUiUtils.executeSafely(
+                () -> {
+                    browser = new Browser(parent, SWT.NONE);
+                    resetBrowser();
+                    browser.addProgressListener(new PitUiUpdatePublisher(browser));
+                    IActionBars actionBars = getViewSite().getActionBars();
+                    IToolBarManager toolBar = actionBars.getToolBarManager();
+                    // create back button
+                    toolBar.add(new Action(BACK_BUTTON_TEXT) {
+                        @Override
+                        public void run() {
+                            browser.back();
+                        }
+                    });
+                    // create home button for navigation
+                    toolBar.add(new Action(HOME_BUTTON_TEXT) {
+                        @Override
+                        public void run() {
+                            if (homeUrlString != null) {
+                                // only set url, if an home is set otherwise NPE
+                                browser.setUrl(homeUrlString);
+                            }
+                        }
+                    });
+                    // create forward button
+                    toolBar.add(new Action(FORWARD_BUTTON_TEXT) {
+                        @Override
+                        public void run() {
+                            browser.forward();
+                    }
+                    });
+                    actionBars.updateActionBars();
+                },
+                parent,
+                "Failed to create the Pit Summary view",
+                "Browser cannot be initialized.");
+    }
+
+    /**
+     * Set URL to blank and reset homeUrl, to have a clean browser after
+     * creation<br>
+     * <b>Needed</b> for testing purposes
```

Comment:
This is not needed anymore by tests, isn't it?
Indeed it's private

Reply:
This is still needed.
It is called while creating and if an `update()` has no file.

Reply:
Directly from the tests? By reflection?

Reply:
No, in the methods `update(...)` and `createPartControl(...)`.